src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -214,16 +214,17 @@ def _get_variance(self, t, predicted_variance=None, variance_type=None):
         # and sample from it to get previous sample
         # x_{t-1} ~ N(pred_prev_sample, variance) == add variance to pred_sample
         variance = (1 - alpha_prod_t_prev) / (1 - alpha_prod_t) * current_beta_t
+        variance = torch.clamp(variance, min=1e-20)
 
         if variance_type is None:
             variance_type = self.config.variance_type
 
         # hacks - were probably added for training stability
         if variance_type == "fixed_small":
-            variance = torch.clamp(variance, min=1e-20)
+            variance = variance
         # for rl-diffuser https://arxiv.org/abs/2205.09991
         elif variance_type == "fixed_small_log":
-            variance = torch.log(torch.clamp(variance, min=1e-20))
+            variance = torch.log(variance, min=1e-20)
```

Comment:
Do we still need the `min` here?

Reply:
ah thanks!

Reply:
https://github.com/huggingface/diffusers/pull/3006/commits/11fbf66e373e425b09649ce14251eee028289201